File: src/main/resources/output/IFC4X3_js/Entities/IfcPropertyAbstraction/IfcProperty/IfcSimpleProperty/IfcPropertyEnumeratedValue/IfcPropertyEnumeratedValue.js
class IfcPropertyEnumeratedValue extends IfcSimpleProperty {
    constructor() {
        /** @type {IFCVALUE[]} */
        this.EnumerationValues = null;
        /** @type {IFCPROPERTYENUMERATION} */
        this.EnumerationReference = null;
    }

    // === WHERE CLAUSES ===
    // WR21 : NOT(EXISTS(EnumerationReference)) OR  NOT(EXISTS(EnumerationValues)) OR  (SIZEOF(QUERY(temp <* EnumerationValues | temp IN EnumerationReference.EnumerationValues)) = SIZEOF(EnumerationValues))
}

File: output/ifc4x3_dev_73740fe4/entity/IfcPropertyEnumeratedValue.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


import java.util.List;

public class IfcPropertyEnumeratedValue extends IfcSimpleProperty {

    public IfcPropertyEnumeration EnumerationReference; // OPTIONAL
    // INVERSE attributes:
    // WR21 : NOT(EXISTS(EnumerationReference))
    // OR  NOT(EXISTS(EnumerationValues))
    // OR  (SIZEOF(QUERY(temp <* EnumerationValues |
    // temp IN EnumerationReference.EnumerationValues))
    // = SIZEOF(EnumerationValues));

    // WHERE constraints:
    // WR21 : NOT(EXISTS(EnumerationReference))
    // OR  NOT(EXISTS(EnumerationValues))
    // OR  (SIZEOF(QUERY(temp <* EnumerationValues |
    // temp IN EnumerationReference.EnumerationValues))
    // = SIZEOF(EnumerationValues));
}
